docs/source/formulations_example.rst

```diff
@@ -0,0 +1,648 @@
+Whew!
+That's a lot, which is why this is best done with additional tooling, whether in code or in the GUI.
+
+A rendering of this example material history is shown below.
+
+.. figure:: _static/GEMD_history_example.png
+    :align: center
+
+    Material History for Blended Margarita B
+
+Repeating this process once for each margarita sample, we can build up a rich dataset for machine learning.
+
+Building a Table
+----------------
+
+We now build a GEM Table to represent the margaritas' material histories in tabular format.
+This table will be used as training data when building a machine learning graphical model.
+For more detailed information on GEM Tables, see the section on :doc:`data extraction <data_extraction>`.
+
+In order to make a Gem Table, we start with a Table Configuration object.
+In this example we will build up the configuration in small steps.
+As we will see, templates are crucial to configuring the table.
+
+The code below defines the rows and defines one column that contains the identifier of each row.
+
+.. code-block:: python
+
+    from citrine.gemtables.rows import MaterialRunByTemplate
+    from citrine.gemtables.variables import RootIdentifier
+    from citrine.gemtables.columns import IdentityColumn
+    from gemd.entity.link_by_uid import LinkByUID
+
+
+    material_templates_to_include = [
+        blended_margarita_template, simple_syrup_template, base_material_template
+    ]
+    scope = "margaritas-id"
+
+    config = TableConfig(
+        name="margarita formulations table",
+        description="",
+        datasets=[dataset.uid],
+        rows=[
+            MaterialRunByTemplate(
+                templates=[LinkByUID.from_entity(t) for t in material_templates_to_include]
+            )
+        ]
+        variables=[RootIdentifier(name="name", headers=["name"], scope=scope)],
+        columns=[IdentityColumn(data_source="name")]
+    )
+
+Let's step through the pieces of this code.
+We gave the configuration a name and description, for human-readability.
+We defined the datasets that contain the material histories; for the purposes of this example we assume everything is in the dataset ``dataset``.
+We then defined the rows as being based on all materials that link to one of a set of templates.
+For the purposes of this example, assume that we have defined separate material templates for the base materials, the simple syrups, the margaritas, and the blended margaritas.
+Notice that we do _not_ include the material template for the unblended margaritas.
+For the purposes of machine learning, we want to compress the mixing and blending into a single training row.
+
+We define a single variable/column that contains a unique identifier for each row.
+Every object on the Citrine Platform has a unique id with scope ``id``, but these are difficult for a human to read.
+In this example we assume that we have given the materials human-readable uniquely identifying names with scope "margaritas-id."
```

Comment:
So why didn't we do that above?  There's no demonstration of assigning a UID.

Reply:
I added custom ids